pkg/security/podsecuritypolicy/provider.go

```diff
@@ -329,3 +338,20 @@ func (s *simpleProvider) isValidHostPort(port int) bool {
 func (s *simpleProvider) GetPSPName() string {
 	return s.psp.Name
 }
+
+// isHostPathAllowed returns no error if the host path is a prefix of any of the
+// allowed host paths allowed by the PSP.
+func isHostPathAllowed(hostPath string, allowedPaths []string) error {
+	// If no allowed paths are specified then allow any path
+	if len(allowedPaths) == 0 {
+		return nil
+	}
+
+	for _, allowedPath := range allowedPaths {
+		if strings.HasPrefix(hostPath, allowedPath) {
```

Comment:
this logic needs to check for exact matches or prefix matches with a trailing `/`